server/vcs/provider/github/converter/comment_event.go

```diff
@@ -37,14 +44,96 @@ func (e CommentEventConverter) Convert(comment *github.IssueCommentEvent) (event
 		eventTimestamp = *comment.Comment.CreatedAt
 	}
 
+	ghPull, err := e.PullGetter.GetPullRequest(baseRepo, pullNum)
+	if err != nil {
+		return event.Comment{}, errors.Wrap(err, "getting pull from github")
+	}
+
+	pull, _, headRepo, err := e.ParseGithubPull(ghPull)
+	if err != nil {
+		return event.Comment{}, errors.Wrap(err, "converting pull request type")
+	}
+
 	return event.Comment{
-		BaseRepo:      baseRepo,
-		MaybeHeadRepo: nil,
-		MaybePull:     nil,
-		User:          user,
-		PullNum:       pullNum,
-		VCSHost:       models.Github,
-		Timestamp:     eventTimestamp,
-		Comment:       comment.GetComment().GetBody(),
+		BaseRepo:  baseRepo,
+		HeadRepo:  headRepo,
+		Pull:      pull,
+		User:      user,
+		PullNum:   pullNum,
+		VCSHost:   models.Github,
+		Timestamp: eventTimestamp,
+		Comment:   comment.GetComment().GetBody(),
 	}, nil
 }
+
+// ParseGithubPull parses the response from the GitHub API endpoint (not
+// from a webhook) that returns a pull request.
+func (e CommentEventConverter) ParseGithubPull(pull *github.PullRequest) (pullModel models.PullRequest, baseRepo models.Repo, headRepo models.Repo, err error) {
+	commit := pull.Head.GetSHA()
+	if commit == "" {
+		err = errors.New("head.sha is null")
+		return
+	}
+	url := pull.GetHTMLURL()
+	if url == "" {
+		err = errors.New("html_url is null")
+		return
+	}
+	headBranch := pull.Head.GetRef()
+	if headBranch == "" {
+		err = errors.New("head.ref is null")
+		return
+	}
+	baseBranch := pull.Base.GetRef()
+	if baseBranch == "" {
+		err = errors.New("base.ref is null")
+		return
+	}
+
+	authorUsername := pull.User.GetLogin()
+	if authorUsername == "" {
+		err = errors.New("user.login is null")
+		return
+	}
+	num := pull.GetNumber()
+	if num == 0 {
+		err = errors.New("number is null")
+		return
+	}
+
+	baseRepo, err = e.ParseGithubRepo(pull.Base.Repo)
+	if err != nil {
+		return
+	}
+	headRepo, err = e.ParseGithubRepo(pull.Head.Repo)
+	if err != nil {
+		return
+	}
+
+	pullState := models.ClosedPullState
+	closedAt := pull.GetClosedAt()
+	updatedAt := pull.GetUpdatedAt()
+	if pull.GetState() == "open" {
+		pullState = models.OpenPullState
+	}
+
+	pullModel = models.PullRequest{
+		Author:     authorUsername,
+		HeadBranch: headBranch,
+		HeadCommit: commit,
+		URL:        url,
+		Num:        num,
+		State:      pullState,
+		BaseRepo:   baseRepo,
+		BaseBranch: baseBranch,
+		ClosedAt:   closedAt,
+		UpdatedAt:  updatedAt,
+	}
+	return
+}
+
+// ParseGithubRepo parses the response from the GitHub API endpoint that
+// returns a repo into the Atlantis model.
+func (c *CommentEventConverter) ParseGithubRepo(ghRepo *github.Repository) (models.Repo, error) {
+	return models.NewRepo(models.Github, ghRepo.GetFullName(), ghRepo.GetCloneURL(), c.RepoConverter.GithubUser, c.RepoConverter.GithubToken)
+}
```

Comment:
Why are you rewriting this? this already exists in the converters package https://github.com/lyft/atlantis/blob/6989c17c90b4b45e7be509bac71010ad254d8fdd/server/vcs/provider/github/converter/pull.go#L16-L17

Reply:
Didn't realize it 😅! I removed it and reused the existing methods